routers/web/user/profile.go

```diff
@@ -70,6 +70,8 @@ func userProfile(ctx *context.Context) {
 		}
 		ctx.Data["HeatmapData"] = data
 		ctx.Data["HeatmapTotalContributions"] = activities_model.GetTotalContributionsInHeatmap(data)
+		showHeatmap := len(data) > 0 || !ctx.ContextUser.KeepActivityPrivate
```

Comment:
ps: plz pay attention that there's another PR will affect the condition here.  #29306

Reply:
Thank you, I will make adjustments accordingly in the future.